bin/make_cfgs.py

```diff
@@ -0,0 +1,182 @@
+sql_line_prefix = 'INSERT INTO nodes ' \
+                  '(name, machine_type, mac_address, is_vm, ' \
+                  'locked, arch, description) VALUES '
+
+parser.add_argument('-c', dest='config_file', required=True,
+                    help='path to configuration file')
+
+parser.add_argument('-d', dest='target_dir', required=True,
+                    help='target directrory for generating files')
+
+args = parser.parse_args()
+
+network_manager.data_file = args.config_file
+nodes = network_manager.get_nodes()
+try:
+    os.mkdir(args.target_dir)
+except FileExistsError:
+    # it is ok if dir already exists
+    pass
+
+target_dir_prefix = args.target_dir + "/"
+
+bmc_user = network_manager.get_option('ipmi_user')
+bmc_pass = network_manager.get_option('ipmi_pass')
+domain = network_manager.get_option('domain')
+
+
+consoles = ""
+roster_recors = ""
+# ----------- iterating and preparing data ---------------
+pxe_node_lines = []
+pxe_node_ptr_lines = []
+nodes_sql_lines = []
+
+for n in nodes:
+    print('Process node' + n['node'])
+    
+    if n['ip_type'] == 'dynamic':
+        pxe_node_lines.append("dhcp-host={},{},{}"
+                          .format(n['mac'], hn(n['node']), n['ip']))
+        pxe_node_ptr_lines.append("ptr-record={}.in-addr.arpa,{}.{}"
+                              .format(reorder_ip(n['ip']),
+                                      hn(n['node']),
+                                      domain))
+
+    if n['bmc_ip_type'] == 'dynamic':
+        pxe_node_lines.append("dhcp-host={},{}-bmc,{}"
+                          .format(n['bmc_mac'], hn(n['node']), n['bmc_ip']))
+        pxe_node_ptr_lines.append("ptr-record={}.in-addr.arpa,{}-bmc.{}"
+                              .format(reorder_ip(n['bmc_ip']),
+                                      hn(n['node']),
+                                      domain
+                                      ))
+
+    if ('hsm_ip_type' in n.keys()) and (n['hsm_ip_type'] == 'dynamic'):
+        pxe_node_lines.append("dhcp-host={},{}-hsm,{}"
+                          .format(n['hsm_mac'], hn(n['node']), n['hsm_ip']))
+        pxe_node_ptr_lines.append("ptr-record={}.in-addr.arpa,{}-hsm"
+                              .format(reorder_ip(n['hsm_ip']), hn(n['node'])))
+
+    consoles = consoles +\
+               "\nCONSOLE name=\"{}\" IPMIOPTS=\"U:{},P:{}\" dev=\"ipmi:{}\""\
+                   .format(hn(n['node']), bmc_user, bmc_pass, n['bmc_ip'])
+    roster_recors = roster_recors +\
+                    n['node']+":\n" +\
+                    "  host: " + n['ip'] + "\n" +\
+                    "  user: root\n"
+    if 't_exclude' in n.keys() and n['t_exclude'] != 'yes':
+        sql_update_body = [
+                "'" + n['node'] + "'",
+                "'" + n['t_machine_type'] + "'",
+                "'" + n['mac'] + "'",
+                "'f'",
+                "'f'",
+                "'x86-64'",
+                "'" + n['comment'] + "'",
+            ]
```

Comment:
Exactly here is the best place to use `.format`

Reply:
I don't see how '.format' could help.

Reply:
replaced to %s